app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -64,12 +66,12 @@ def feeds
     end
 
     def machine_id(feed)
-      os_resource_for(feed).properties['Machine Id']
+      os_resource_for(hawk_escape_id(feed)).properties['Machine Id']
     end
 
     def os_resource_for(feed)
       with_provider_connection do |connection|
-        os = os_for(feed)
+        os = os_for(hawk_escape_id(feed))
```

Comment:
To verify on the UI, you'll also need "the other side" of the link, ie, a RHEV-M instance. We have both on torii, so, you could try adding them on your local MiQ instance and see if it still works. I remember theute mentioning that there was a trick to get it working properly. Perhaps @abonas would know about such trick?